mm2src/coins/solana.rs

```diff
@@ -378,6 +396,246 @@ impl SolanaCoin {
         let guard = self.spl_tokens_infos.lock().unwrap();
         (*guard).clone()
     }
+
+    fn send_hash_time_locked_payment(&self, args: SendPaymentArgs<'_>) -> SolTxFut {
+        let receiver = Pubkey::new(args.other_pubkey.iter().as_slice());
+        let swap_program_id = Pubkey::new(
+            try_tx_fus_opt!(
+                args.swap_contract_address.as_ref(),
+                format!(
+                    "Unable to extract Bytes from args.swap_contract_address ( {:?} )",
+                    args.swap_contract_address
+                )
+            )
+            .as_slice(),
+        );
```

Comment:
```suggestion
        let swap_program_id = Pubkey::new(try_tx_fus_opt!(
            args.swap_contract_address,
            format!(
                "Unable to extract Bytes from args.swap_contract_address ( {:?} )",
                args.swap_contract_address
            )
        ));
```
Please find any other instances that uses as_slice when it's not needed and fix them.

Reply:
I’ve reviewed the entire solana.rs file. Hopefully, I didn’t miss anything.